lib/p2p/Pool.ts

```diff
@@ -669,6 +669,14 @@ class Pool extends EventEmitter {
     return peer;
   }
 
+  public tryGetPeer = (peerPubKey: string) => {
+    try {
+      return this.getPeer(peerPubKey);
+    } catch (err) {
+      return;
```

Comment:
nit: maybe we should consider adding a debug/trace logging here to make future debugging easier?

Reply:
If we want to log this, I think it would be better to do it in an outer layer, for having a better context. For example, in `swaps.failDeal`, by adding an `else` condition here:

https://github.com/ExchangeUnion/xud/blob/8d398622612e6b0f70f4aa7d2a98620627b1f8a3/lib/swaps/Swaps.ts#L1132-L1140

Also, I looked again in `pool.getPeer` usages throughout the codebase, and although all of them might encounter an exception being thrown (peer being disconnected), the error catch is covered by some function in the stack, so xud won't crash. But I think it worth checking it more seriously, and perhaps do better job with the logs. 

I'll open a separate issue for that (don't want to delay this PR from being merged).